.github/workflows/cont_integration.yml

```diff
@@ -4,141 +4,137 @@ name: CI
-      - name: Install Rust toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-            toolchain: stable
-            override: true
-            profile: minimal
-            target: "wasm32-unknown-unknown"
-      - name: Rust Cache
-        uses: Swatinem/rust-cache@v2.2.1
-      - name: Check bdk
-        working-directory: ./crates/bdk
-        run: cargo check --target wasm32-unknown-unknown --no-default-features --features bitcoin/no-std,miniscript/no-std,bdk_chain/hashbrown,dev-getrandom-wasm
-      - name: Check esplora
-        working-directory: ./crates/esplora
-        run: cargo check --target wasm32-unknown-unknown --no-default-features --features bitcoin/no-std,miniscript/no-std,bdk_chain/hashbrown,async
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@v5
+      # Cache Nix artifacts
+      - uses: DeterminateSystems/magic-nix-cache-action@v2
+      # Rustfmt
+      - name: Rustfmt
+        run: nix build -L .#ci.fmt --keep-failed
 
-  fmt:
-    name: Rust fmt
+  clippy:
+    name: clippy
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Install Rust toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-            toolchain: stable
-            override: true
-            profile: minimal
-            components: rustfmt
-      - name: Check fmt
-        run: cargo fmt --all -- --config format_code_in_doc_comments=true --check
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@v5
+      # Cache Nix artifacts
+      - uses: DeterminateSystems/magic-nix-cache-action@v2
+      # Clippy
+      - name: Clippy
+        run: nix build -L .#ci.clippy --keep-failed
 
-  clippy_check:
+  test-latest:
+    name: Build, test, check latest Rust
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1
-      - uses: actions-rs/toolchain@v1
-        with:
-            # we pin clippy instead of using "stable" so that our CI doesn't break
-            # at each new cargo release
-            toolchain: "1.67.0"
-            components: clippy
-            override: true
-      - name: Rust Cache
-        uses: Swatinem/rust-cache@v2.2.1
-      - uses: actions-rs/clippy-check@v1
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          args: --all-features --all-targets -- -D warnings
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@v5
+      # Cache Nix artifacts
+      - uses: DeterminateSystems/magic-nix-cache-action@v2
+      # Cache Builds
+      - name: Build and Cache latest Rust
+        run: nix build -L . --keep-failed
+      # Tests: latest
+      - name: Test workspace all-features latest
```

Comment:
**suggestion:** Nix could build all of these in parallel, if you bundle them all into one derivation. AFAIK usually done with something like `pkgs.linkFarmFromDrvs` [example](https://github.com/rustshop/flakebox/blob/5dea81cad2be257a2cbccbd42b2f1c0fdf69008d/checks/workspace-cross-compile/default.nix#L93)

Reply:
Yes, I will open an issue to track these enhancements: #1185.

Reply:
Tracked in #1185.